code/modules/library/computers/accounting.dm

```diff
@@ -0,0 +1,143 @@
+/obj/machinery/computer/accounting
+	name = "accounting computer"
+	desc = "For virtual bookkeeping. This computer is used in the calculation of overages."
+	pass_flags = PASSTABLE
+	machine_flags = WRENCHMOVE | FIXED2WORK
+	icon = 'icons/obj/library.dmi'
+	icon_state = "computer"
+	req_access = list(access_library)
+	var/savings = 0 //Total savings this cycle
+	var/oldgood = -1 //Was our last calculation correct? -1 indicates no last query this cycle, resets to -1 at cycle end, 0 is bad, 1 is good
+	var/oldentry = 0 //Our last attempted difference
+
+	var/minuend
+	var/subtrahend
+	var/oldquery
+	var/currentqueryname
+	var/nextmin
+	var/nextsub
+	var/entryinfo = "<big>Accounting Report</big><BR>"
+	var/list/contributors = list()
+
+/obj/machinery/computer/accounting/New()
+	..()
+	generatenext()
+	readynext()
+
+/obj/machinery/computer/accounting/Destroy()
+	contributors.Cut()
+	..()
+
+/obj/machinery/computer/accounting/attack_hand(mob/user as mob)
+	if(..())
+		return
+	if(stat & (NOPOWER|BROKEN|FORCEDISABLE))
+		return
+	ui_interact(user)
+
+/obj/machinery/computer/accounting/ui_interact(mob/user, ui_key="main", datum/nanoui/ui=null, var/force_open=NANOUI_FOCUS)
+	user.set_machine(src)
+
+	var/data[0]
+	data["src"] = "\ref[src]"
+	data["savings"] = savings
+	data["oldgood"] = oldgood
+	data["oldquery"] = oldquery
+	data["oldentry"] = oldentry
+	data["minuend"] = minuend
+	data["subtrahend"] = subtrahend
+	data["currentqueryname"] = currentqueryname
+	data["nextquery"] = "[nextmin] - [nextsub]"
+	ui = nanomanager.try_update_ui(user, src, ui_key, ui, data, force_open)
+	if (!ui)
+		ui = new(user, src, ui_key, "accounting.tmpl", src.name, 600, 500)
+		ui.set_initial_data(data)
+		ui.open()
+
+/obj/machinery/computer/accounting/Topic(href, href_list)
+	if(..())
+		return 1
+	if(!allowed(usr))
+		to_chat(usr, "<span class='warning'>You do not have accounting clearance.</span>")
+		return
+	if(href_list["reg"])
+		if(!in_range(src, usr))
+			return 1
+		var/sol = text2num(href_list["reg"])
+		if(isnull(sol))
+			visible_message("<span class='warning'>[src] buzzes rudely.</span>")
+			playsound(src, 'sound/machines/buzz-two.ogg', 50, 0)
+			return 1
+		solve_query(sol)
+		if(!(usr in contributors))
+			contributors += usr
```

Comment:
```suggestion
		contributors |= usr
```

You should consider storing just the names rather than references to mobs, you only need this to print their names.

Reply:
Ah yes, I had a reason for this originally in terms of pulling out their bank account, but I think it is no longer necessary since I send a wallet with money